--- //depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-08-26 07:57:01.000000000 -0500
+++ /Users/achhatre/Perforce/depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-08-26 07:57:01.000000000 -0500
@@ -354,7 +354,7 @@
             visible_toolbar_buttons: {
                 'emoticons': true,
                 'call': false,
-                'clear': true,
+                'clear': false,
                 'toggle_participants': true
             },
             websocket_url: undefined,
@@ -429,8 +429,8 @@
 
         var HEADER_WEIGHTS = {};
         HEADER_WEIGHTS[HEADER_CURRENT_CONTACTS]    = 0;
-        HEADER_WEIGHTS[HEADER_UNGROUPED]           = 1;
-        HEADER_WEIGHTS[HEADER_REQUESTING_CONTACTS] = 2;
+        HEADER_WEIGHTS[HEADER_REQUESTING_CONTACTS] = 1;
+        HEADER_WEIGHTS[HEADER_UNGROUPED]           = 2;
         HEADER_WEIGHTS[HEADER_PENDING_CONTACTS]    = 3;
 
         // Module-level variables
@@ -1241,7 +1241,7 @@
                     .html(converse.templates.chatbox(
                             _.extend(this.model.toJSON(), {
                                     show_toolbar: converse.show_toolbar,
-                                    label_personal_message: __('Personal message')
+                                    label_personal_message: ZmMsg.chatTypeMessage
                                 }
                             )
                         )
@@ -1304,7 +1304,7 @@
                 }
                 var message = template({
                     'sender': msg_dict.sender,
-                    'time': msg_time.format('hh:mm'),
+                    'time': msg_time.format('hh:mma').slice(0,-1), //TODO handle locale specific formats here.
                     'username': username,
                     'message': '',
                     'extra_classes': extra_classes
@@ -1878,7 +1878,8 @@
                     label_logout: __('Log out'),
                     allow_logout: converse.allow_logout
                 });
-                this.$tabs.append(converse.templates.contacts_tab({label_contacts: LABEL_CONTACTS}));
+                // ZCS - modified template
+                //this.$tabs.append(converse.templates.contacts_tab({label_contacts: LABEL_CONTACTS}));
                 if (converse.xhr_user_search) {
                     markup = converse.templates.search_contact({
                         label_contact_name: __('Contact name'),
@@ -3373,7 +3374,8 @@
                         'fullname': fullname,
                         'image_type': roster_item.get('image_type'),
                         'image': roster_item.get('image'),
-                        'url': roster_item.get('url')
+                        'url': roster_item.get('url'),
+                        'chat_status': roster_item.get('chat_status')
                     });
                 }
                 if (msgid && chatbox.messages.findWhere({msgid: msgid})) {
@@ -4959,9 +4961,11 @@
                     options_list = [],
                     that = this;
                 this.$el.html(converse.templates.choose_status());
+
+                // ZCS - modified the rendering of status from I am <status> to show only the <status>
                 this.$el.find('#fancy-xmpp-status-select')
                         .html(converse.templates.chat_status({
-                            'status_message': this.model.get('status_message') || __("I am %1$s", this.getPrettyStatus(chat_status)),
+                            'status_message': this.model.get('status_message') || __("%1$s", this.getPrettyStatus(chat_status)),
                             'chat_status': chat_status,
                             'desc_custom_status': __('Click here to write a custom status message'),
                             'desc_change_status': __('Click to change your chat status')
@@ -5010,6 +5014,8 @@
                     converse.logOut();
                 } else {
                     this.model.setStatus(value);
+                    // Bug fix: Reset the status message whenever status is toggled from drop down
+                    this.model.setStatusMessage('');
                     this.$el.find(".dropdown dd ul").hide();
                 }
             },
@@ -5034,7 +5040,8 @@
                 var stat = model.get('status');
                 // # For translators: the %1$s part gets replaced with the status
                 // # Example, I am online
-                var status_message = model.get('status_message') || __("I am %1$s", this.getPrettyStatus(stat));
+                // ZCS - modified the rendering of status from I am <status> to show only the <status>
+                var status_message = model.get('status_message') || __("%1$s", this.getPrettyStatus(stat));
                 this.$el.find('#fancy-xmpp-status-select').html(
                     converse.templates.chat_status({
                         'chat_status': stat,
