--- //depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-09-08 10:29:13.000000000 -0500
+++ /Users/achhatre/Perforce/depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-09-08 10:29:13.000000000 -0500
@@ -4291,9 +4291,11 @@
                 } else if (presence_type === 'unavailable' && contact) {
                     // Only set the user to offline if there aren't any
                     // other resources still available.
-                    if (contact.removeResource(resource) === 0) {
+
+                    // TODO: Bug within converse. https://github.com/jcbrand/converse.js/issues/487
+                    //if (contact.removeResource(resource) === 0) {
                         contact.save({'chat_status': "offline"});
-                    }
+                    //}
                 } else if (contact) { // presence_type is undefined
                     this.addResource(bare_jid, resource);
                     contact.save({'chat_status': chat_status});
