--- //depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.1/converse.js	2015-08-12 09:31:50.000000000 -0500
+++ /Users/achhatre/Perforce/depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.1/converse.js	2015-08-12 09:31:50.000000000 -0500
@@ -3306,6 +3306,16 @@
                      * the user, then a chat box is created here which then
                      * opens automatically :(
                      */
+
+                    /* 
+                        Fix for Bug: 100630 - This is fixed in converse version 0.9.2
+                        Note - The below if block need not be migrated as part of the
+                        refactor process since this was fixed as a part of converse v0.9.2.
+                    */
+                    if ($message.find('body').length === 0) {
+                        return true;
+                    }
+
                     var fullname = roster_item.get('fullname');
                     fullname = _.isEmpty(fullname)? contact_jid: fullname;
                     chatbox = this.create({
@@ -3314,7 +3324,8 @@
                         'fullname': fullname,
                         'image_type': roster_item.get('image_type'),
                         'image': roster_item.get('image'),
-                        'url': roster_item.get('url')
+                        'url': roster_item.get('url'),
+                        'chat_status': roster_item.get('chat_status')
                     });
                 }
                 if (msgid && chatbox.messages.findWhere({msgid: msgid})) {
