--- //depot/zimbra/users/achhatre/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-09-29 09:46:09.000000000 -0500
+++ /Users/rshah/projects/p4/chat/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-09-29 09:46:09.000000000 -0500
@@ -1237,11 +1237,13 @@
             },
 
             render: function () {
+                var contact = converse.roster.get(this.model.get('jid'));
                 this.$el.attr('id', this.model.get('box_id'))
                     .html(converse.templates.chatbox(
                             _.extend(this.model.toJSON(), {
                                     show_toolbar: converse.show_toolbar,
-                                    label_personal_message: ZmMsg.chatTypeMessage
+                                    label_personal_message: ZmMsg.chatTypeMessage,
+                                    chat_status: contact ? contact.get('chat_status') : "online"
                                 }
                             )
                         )