--- //depot/zimbra/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-12-09 07:13:08.000000000 -0600
+++ /Users/rshah/projects/p4/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-12-09 07:13:08.000000000 -0600
@@ -692,8 +692,9 @@
                 } else {
                     document.title = document.title.replace(/^Messages \(\d+\) /, "Messages (" + this.msg_counter + ") ");
                 }
-                window.blur();
-                window.focus();
+                //ZCS - Do not focus on the window. On chrome/safari the focus does not stay on html box. See bug 101949.
+                /*window.blur();
+                window.focus();*/
             } else if (document.title.search(/^Messages \(\d+\) /) != -1) {
                 document.title = document.title.replace(/^Messages \(\d+\) /, "");
             }
