--- //depot/zimbra/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-12-24 08:46:00.000000000 -0600
+++ /Users/rshah/projects/p4/JUDASPRIEST/ZimbraWebClient/converse/converse.js-0.9.4/converse.js	2015-12-24 08:46:00.000000000 -0600
@@ -6032,7 +6032,7 @@
         'contacts': {
             'get': function (jids) {
                 var _transform = function (jid) {
-                    var contact = converse.roster.get(Strophe.getBareJidFromJid(jid));
+                    var contact = converse.roster && converse.roster.get(Strophe.getBareJidFromJid(jid));
                     if (contact) {
                         return contact.attributes;
                     }
